### DISCUSSION TITLE
PWM Frequency per Channel or GPIO

### DISCUSSION THREAD

**Discussion by phanky5** [arendst/Tasmota#19503] [Q&A]
I would like to control several devices via PWM. I need different PWM frequencies for each device. As far as I can see there is only a global setting for PWM frequency in Tasmota using the command PWMFrequency in the console.

Is there a way to set different frequencies for different PWM channel or per GPIO used?

**Comment by sfromis** [comment]
No, only one PWM frequency is available.

**Comment by DoeWayne** [comment]
On ESP32 you can use PWMFrequency Hz,Channel now you set different frequency for the mentioned channel

But how can this be saved

**Comment by Jason2866** [comment]
Via `autoxec.bat` provide the settings you want.
